Document ConditionalApply and AssertSameNode

Mats-SX opened the pull request:
- Also added AntiConditionalApply
- Moved over to the new manual infrastructure
- Improved some textual descriptions and formatting

Builds upon https://github.com/neo4j/neo4j/pull/5907

Mats-SX committed:
Document ConditionalApply and AssertSameNode

- Also added AntiConditionalApply
- Moved over to the new manual infrastructure
- Improved some textual descriptions and formatting

pontusmelke commented:
https://github.com/neo4j/neo4j/pull/5907 has been rebased, you probably need to cherry pick the last commit on top of that

Mats-SX commented:
I'll just wait until it is merged, and then cherry-pick this commit on top of the latest HEAD and update this PR.

Mats-SX committed:
Add necessary capabilities to documentation API

We can now run queries with `EXPLAIN` and insert the produced plan description.
